app/src/main/java/org/dicekeys/app/fragments/dicekey/SoloKeyFragment.kt

```diff
@@ -1,17 +1,59 @@
 package org.dicekeys.app.fragments.dicekey
 
+import android.content.Context
+import android.hardware.usb.UsbManager
 import android.os.Bundle
 import android.view.View
+import androidx.core.widget.doAfterTextChanged
+import androidx.fragment.app.viewModels
 import dagger.hilt.android.AndroidEntryPoint
-import org.dicekeys.app.AppFragment
+import org.dicekeys.app.Application
 import org.dicekeys.app.R
 import org.dicekeys.app.databinding.SolokeyFragmentBinding
-import org.dicekeys.app.viewmodels.DiceKeyViewModel
+import org.dicekeys.app.extensions.dialog
+import org.dicekeys.app.extensions.errorDialog
+import org.dicekeys.app.viewmodels.SoloKeyViewModel
+import javax.inject.Inject
 
 @AndroidEntryPoint
-class SoloKeyFragment: AbstractDiceKeyFragment<SolokeyFragmentBinding>(R.layout.solokey_fragment) {
+class SoloKeyFragment : AbstractDiceKeyFragment<SolokeyFragmentBinding>(R.layout.solokey_fragment) {
+
+    @Inject
+    lateinit var viewModelFactory: SoloKeyViewModel.AssistedFactory
+
+    private val soloKeyViewModel: SoloKeyViewModel by viewModels {
+        SoloKeyViewModel.provideFactory(viewModelFactory, requireActivity().application as Application, requireContext().getSystemService(Context.USB_SERVICE) as UsbManager, viewModel.diceKey)
+    }
+
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
+        binding.diceKeyVM = viewModel
+        binding.vm = soloKeyViewModel
+
+
+        binding.btnDown.setOnClickListener { soloKeyViewModel.sequencUpDown(false) }
+        binding.btnUp.setOnClickListener { soloKeyViewModel.sequencUpDown(true) }
+        binding.etSequenceNumber.doAfterTextChanged { edittext ->
+            try{
+                soloKeyViewModel.updateSequence(edittext.toString().toInt())
```

Comment:
Documentation please.
````
// Update the sequence number in the model when the text field's recipe changes
```

Does this change on every character (it should) or only when the user closes the field?

Reply:
It's a TextChangeListener os the change is instant

Reply:
Okay.  Worth documenting since the onChanged event in the W3 DOM sends changed events only after the user leaves the field